src/net/rpc.rs

```diff
@@ -1509,6 +1510,91 @@ impl ConversationHttp {
         return response.send(http, fd).map(|_| ());
     }
 
+    /// Handle a GET health.
+    /// If the node's height is not up to the max height of all of its bootstrap nodes, we return
+    /// false for the health check.
+    /// Additionally return what percent of the max height the node height is at.
+    fn handle_get_health<W: Write>(
+        http: &mut StacksHttp,
+        fd: &mut W,
+        req: &HttpRequestType,
+        network_id: u32,
+        sortdb: &SortitionDB,
+        chainstate: &StacksChainState,
+        peerdb: &PeerDB,
+        block_stats: Option<&HashMap<NeighborKey, NeighborBlockStats>>,
+    ) -> Result<(), net_error> {
+        let response_metadata = HttpResponseMetadata::from(req);
+
+        let block_height = match chainstate.get_stacks_chain_tip(sortdb)? {
```

Comment:
Instead of hitting the DB, should we be using the caching layer introduced by Jude in https://github.com/blockstack/stacks-blockchain/pull/2746 ?

Reply:
I would get that we would want an un-cached version of block height. This value is ultimately used in the comparison `block_height >= (max_height - 1)`, which determines whether or not the node returns a 200 level response or not.